polars/polars-core/src/frame/mod.rs

```diff
@@ -2261,6 +2261,126 @@ impl DataFrame {
+    /// Try in keep output similar to pandas
+    ///
+    /// # Example
+    ///
+    /// ```no_run
+    /// # use polars_core::prelude::*;
+    /// let df1: DataFrame = df!("categorical" => &["d","e","f"],
+    ///                          "numeric" => &[1, 2, 3],
+    ///                          "object" => &["a", "b", "c"])?;
+    /// assert_eq!(df1.shape(), (3, 3));
+    ///
+    /// let df2: DataFrame = df1.describe(None);
+    /// assert_eq!(df2.shape(), (8, 4));
+    /// println!("{}", df2);
+    /// # Ok::<(), PolarsError>(())
+    /// ```
+    ///
+    /// Output:
+    ///
+    /// ```text
+    /// shape: (8, 4)
+    /// ┌──────────┬─────────────┬─────────┬────────┐
+    /// │ describe ┆ categorical ┆ numeric ┆ object │
+    /// │ ---      ┆ ---         ┆ ---     ┆ ---    │
+    /// │ str      ┆ str         ┆ f64     ┆ str    │
+    /// ╞══════════╪═════════════╪═════════╪════════╡
+    /// │ count    ┆ null        ┆ 3.0     ┆ null   │
+    /// ├╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┤
+    /// │ mean     ┆ null        ┆ 2.0     ┆ null   │
+    /// ├╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┤
+    /// │ std      ┆ null        ┆ 1.0     ┆ null   │
+    /// ├╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┤
+    /// │ min      ┆ null        ┆ 1.0     ┆ null   │
+    /// ├╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┤
+    /// │ 0.25%    ┆ null        ┆ 1.5     ┆ null   │
+    /// ├╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┤
+    /// │ 0.5%     ┆ null        ┆ 2.0     ┆ null   │
+    /// ├╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┤
+    /// │ 0.75%    ┆ null        ┆ 2.5     ┆ null   │
+    /// ├╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌┤
+    /// │ max      ┆ null        ┆ 3.0     ┆ null   │
+    /// └──────────┴─────────────┴─────────┴────────┘
+    /// ```
+    #[must_use]
+    pub fn describe(&self, percentiles: Option<&[f64]>) -> Self {
+        fn is_numeric(s: &Series) -> bool {
+            let types = [
+                DataType::Int8,
+                DataType::Int16,
+                DataType::Int32,
+                DataType::Int64,
+                DataType::UInt8,
+                DataType::UInt16,
+                DataType::UInt32,
+                DataType::UInt64,
+                DataType::Float32,
+                DataType::Float64,
+            ];
+
+            types.contains(s.dtype())
+        }
+
+        fn describe_cast(df: &DataFrame) -> DataFrame {
+            let mut columns: Vec<Series> = vec![];
+
+            for s in df.columns.iter() {
+                if is_numeric(s) || *s.dtype() == DataType::Boolean {
+                    columns.push(s.cast(&DataType::Float64).expect("cast to float failed"))
+                } else {
+                    columns.push(s.clone())
+                }
+            }
+
+            DataFrame::new(columns).unwrap()
+        }
+
+        fn count(df: &DataFrame) -> DataFrame {
+            let columns = df.apply_columns_par(&|s| s.count_as_series());
```

Comment:
Simply wrap `Series::new(s.name(), [s.len() as IdxSize])` here.

Reply:
Done these changes but can't do concat_df now 
because column datatypes (dtypes) in the two DataFrames do not match ie 

`left.name='categorical' with left.dtype=u32 != right.dtype=str with right.name='categorical'")' `

As I changed count to always be int where mean etc are sometimes string or a float due to describe_cast

Need to think about how to solve that.